PRINCIPLES.md

```diff
@@ -0,0 +1,259 @@
+- [See Also](#see-also)
+
+## Summary
+
+GitOps is a set of principles for operating and managing software systems.
+
+When using GitOps, the desired state of a system or subsystem is defined declaratively as versioned, immutable data, and the running system's configuration is continuously derived from this data.
+
+GitOps principles were derived from modern software operations but are rooted in pre-existing and widely adopted best practices. These principles are:
+
+1. [**The principle of declarative configuration**](#1-declarative-configuration)
+
+    A system managed by GitOps must have its _Desired State_ expressed declaritively as data in a format writable and readable by both humans and machines.
+
+2. [**The principle of immutable configuration versions**](#2-immutable-configuration-versions)
+
+    _Desired State_ is stored in a way that supports versioning, immutability of versions, and retains a complete version history.
+    We call systems that store desired state in this way _State Stores_.
+
+3. [**The principle of continuous state reconciliation**](#3-continuous-state-reconciliation)
+
+    Software agents continuously, and automatically, compare a system's _Actual State_ to its _Desired State_.
+    If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+    These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+
+4. [**The principle of operations through declaration**](#4-operations-through-declaration)
+
+    The mechanism through which change is applied to a system by either a human operator or another system is through the creation of a new declarative version of the desired state in the state store, not through directly interacting with the running system.
+
+## Introduction
+
+The systems that we manage vary widely; from battery-powered devices driven by microcontrollers to globally distributed systems with millions and sometimes billions of users.
+
+What is common across all systems is the need for human operators to make changes to the running state, it is during change that systems are most at risk of failure.
+
+GitOps is a framework in which all change is applied to a system in a consistent mechanism that both reduces risk and improves the situational awareness of human operators working inside the system.
+
+GitOps can be applied to managing large systems or applied only to the smaller sub-system.
+
+See also the [Rationale for GitOps](RATIONALE.md).
+
+## Scope
+
+GitOps concerns the interaction between humans and technical systems, and between technical systems.
+GitOps is not concerned with processes of human decision making or organisation, only how human decisions about technical systems are recorded and applied.
+It is a structured process through which technical systems can be modified reliably.
+
+GitOps is _not_ intended as a model for judging human organisational designs.
```

Comment:
Not sure how this could even be attempted using GItops?
```suggestion
```

Reply:
For example, keeping an organizations employee reporting structure in git. But I'm not actually 100% sure we should list this as out of scope. For example, certain governmental and legal sources are kept in version control - where changes from those sources could be automatically reconciled to a public facing website using a GitOps software agent. Let's discuss this in the next co-working session If it remains controversial, we could leave it out of the first iteration.

Reply:
CNCF and Kubernetes TOC being an example.